pkg/daemon/ceph/osd/daemon.go

```diff
@@ -84,10 +86,42 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 		if err := releaseLVMDevice(context, volumeGroupName); err != nil {
 			return fmt.Errorf("failed to release device from lvm. %+v", err)
 		}
+
 	}
+
 	return nil
 }
 
+func handleTerminate(context *clusterd.Context, lvPath, volumeGroupName string) error {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	for {
+		select {
+		case <-sigc:
+			logger.Infof("shutdown signal received, exiting...")
+
+			pid, err := context.Executor.ExecuteCommandWithOutput(false, "", "fuser", "-a", lvPath)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve process ID for - %s. Error %+v", lvPath, err)
+			}
+
+			logger.Infof("process ID for logical volume  %s : %s", lvPath, pid)
+			// shut down the osd-ceph process so that lvm release does not show device in use error.
+			if pid != "" {
```

Comment:
We must log the case where pid is empty and fail.

Reply:
> 1. Ideally we should find a solution for freeing the LVs immediately when the pod exits. We don't see this issue with freeing the LVs when we are running c-v directly on a device in /dev, so the question is why we see this issue with OSDs on PVs. The issue must be related to how we copy the device to another location and how we enable LVM in the container. Perhaps we need to look at this approach again and see if there is a cleaner way to enable LVM for use by OSDs on PVs.

Adding the missing dev mounts does not release the LV. So we still need to figure out something for #1 here.  dev mounts only help us in #2

Reply:
I think pid empty could be a valid case here. If its Process ID is empty, we can directly go ahead and deactivate the device using `vgchange` as no process is associated with the device file. If its not empty we should first kill the process and then deactivate the device.  Let me know if I'm missing something here.

Reply:
Makes sense.